File: 0014-longest-common-prefix/0014-longest-common-prefix.c
char* longestCommonPrefix(char** strs, int strsSize) {
    char* prefix = strs[0];
    for(int i = 1; i < strsSize; i ++){
        char* str = strs[i];
        int j;
        for (j = 0; j < strlen(str); j ++){
            if(prefix[j] == '\0') {
                break;
            }
            else if(prefix[j] != str[j]){
                prefix[j] = '\0';
                break;
            }
        }
        if(j == strlen(str) && prefix[j] != '\0'){
            prefix[j] = '\0';
        }
    }
    return prefix;
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        let prefix = &strs[0];
//        let prefix_chars = prefix.chars();
        let mut index: usize = prefix.len();
        let mut j: usize = index;
        for i in strs.iter(){
            if i == prefix {
                continue;
            }
            else{
                if i.len() == 0{
                    return String::from("");
                }
                for (ind, ch) in i.chars().enumerate(){
                    j = ind;
                    if prefix.chars().nth(ind) == Some(ch) {
                        continue;
                    }
                    else{
                        if index > ind {
                            index = ind;
                        }
                        break;
                    }
                    println!("{ind} -> {ch} -> {:?}", (prefix.chars().nth(ind)) == Some(ch));
                }
                if index > j {
                    index = j + 1;
                }
            }
        }
        println!("{j}");
        return String::from(&prefix[..index]);
    }
}